### PR TITLE
Create tooltip

### PR DESCRIPTION
**Pull request by stephenjwatkins**
Creates `<Tooltip />`.

- [x] Code complete
- [x] Unit tests
- [x] Component TSDoc written
- [x] Storybook stories and docs
- [x] Specs and documentation up-to-date
- [x] Add changeset

Other:

- Creates storybook `OverlayLayoutDecorator` for stories with overlays
- Forwards ref on button components for attaching overlays
- Fixes storybook breakpoints parsing
- Fixes incompatibility with vitest timers

### CODE REVIEW

**Comment by OskiTheCoder** [comment]
haven't looked at the PR yet, but does this need a `Minor` changeset or not since its in pre-release?

**Comment by stephenjwatkins** [comment]
> haven't looked at the PR yet, but does this need a `Minor` changeset or not since its in pre-release?

it does. thanks for catching that. i'm going to make a PR template so i don't keep forgetting things

File: easy-ui-react/src/Button/utilities.ts [hunk 1]
```diff
@@ -17,3 +18,16 @@ export function logWarningIfInvalidColorVariantCombination(
     );
   }
 }
+
+/**
+ * Removes aria-specific props from being applied to button DOM element.
+ */
+export function omitReactAriaSpecificProps(props: object) {
```
Comment (stephenjwatkins) [inline easy-ui-react/src/Button/utilities.ts:25]: i don't love this. let me know if you have any better ideas.

the problem is that `useButton(props)` is swallowing some props that need to get passed into the `<button />` element. specifically the `onPointerEnter` and `onPointerLeave` events that are being supplied by the Tooltip. they get passed into `useButton()` but don't come out the other side.

**Comment by ldewald** [comment]
**Question:** What is the expected behavior for hiding/showing when going from hovering on one trigger to another trigger? 

While poking at the Storybook, it seems like the initial mouseover has a delay before displaying the tooltip then subsequent mouseovers display the tooltip immediately. 
tooltip-placement

**Comment by stephenjwatkins** [comment]
> While poking at the Storybook, it seems like the initial mouseover has a delay before displaying the tooltip then subsequent mouseovers display the tooltip immediately.

i was going to note that in my review comment. it has a warmup and cooldown period. it's noted in the Spectrum guidelines here: https://spectrum.adobe.com/page/tooltip/#Warmup-and-cooldown

it's using the default from Aria currently, which is 1.5s on enter and .5s on close, but we can customize that to our own liking. i was going to ask Rachel her thoughts

File: easy-ui-react/src/Tooltip/Tooltip.module.scss [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  @include component-token("tooltip", "shadow", theme-token("shadow.overlay"));
+  @include component-token(
+    "tooltip",
+    "background",
+    theme-token("color.surface.overlay")
+  );
+  @include component-token(
+    "tooltip",
+    "border_radius",
+    design-token("shape.border_radius.base")
```
Comment (OskiTheCoder) [inline easy-ui-react/src/Tooltip/Tooltip.module.scss:15]: **question**: out of curiosity, is there a benefit to wrapping design tokens in component tokens?

**Review by ldewald — APPROVED** [review]
Nice

File: easy-ui-react/src/Button/utilities.ts (hunk 1 above)
Comment (OskiTheCoder) [reply easy-ui-react/src/Button/utilities.ts:25]: i think what you did here is sensible

File: easy-ui-react/src/Tooltip/Tooltip.module.scss [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  );
+  @include component-token("tooltip", "color", theme-token("color.text"));
+
+  background: component-token("tooltip", "background");
+  border-radius: component-token("tooltip", "border_radius");
```
Comment (OskiTheCoder) [inline easy-ui-react/src/Tooltip/Tooltip.module.scss:20]: looks like our design specs need to be updated, ive noticed the `border_radius` discrepancy in a few places

File: easy-ui-react/src/Tooltip/Tooltip.module.scss [hunk 4]
```diff
@@ -0,0 +1,74 @@
+@use "../styles/common" as *;
+
+.Tooltip {
+  @include component-token("tooltip", "arrow-size", "1rem");
+  @include component-token("tooltip", "max-width", 320px);
```
Comment (OskiTheCoder) [inline easy-ui-react/src/Tooltip/Tooltip.module.scss:5]: **question** is this `320px` spec'd somewhere?

**Review by OskiTheCoder — APPROVED** [review]
great work! just one non-blocking question

**Comment by OskiTheCoder** [comment]
this is probably more or less just a storybook thing, but i noticed when i zoomed in and out, the tooltip moved to the left. can't seem to replicate this everytime though, so my assumption is that its more or less just a storybook issue, but thought id mention it just in case 

https://user-images.githubusercontent.com/44451175/235483399-a480ada4-06c4-4f1d-9c8b-644df28722d8.mov

File: easy-ui-react/src/Tooltip/Tooltip.module.scss (hunk 4 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/Tooltip/Tooltip.module.scss]: good catch. it's not. i wanted to guard against tooltip overflow on mobile screens, but i figured out a better way to handle that. this is now removed

File: easy-ui-react/src/Tooltip/Tooltip.module.scss (hunk 3 above)
Comment (stephenjwatkins) [reply easy-ui-react/src/Tooltip/Tooltip.module.scss:19]: ah good call

**Comment by stephenjwatkins** [comment]
> this is probably more or less just a storybook thing, but i noticed when i zoomed in and out, the tooltip moved to the left. can't seem to replicate this everytime though, so my assumption is that its more or less just a storybook issue, but thought id mention it just in case

good catch. this appears to be an Aria bug:

https://github.com/adobe/react-spectrum/issues/3678
https://github.com/adobe/react-spectrum/issues/2700

fortunately it's likely not a showstopper but something we should keep an eye on

**Pull request merged**